### PR TITLE
solution

### PR DESCRIPTION
**Pull request by GoldenSpade**
[DEMO LINK](https://billizane.github.io/react_static-list-of-todos/)

### CODE REVIEW

**Review by stas-dukach — APPROVED** [review]
Gj

File: src/components/Todo/Todo.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+const Todo = ({ todo }) => (
+  <div className="container">
+    <User user={todo.user} />
+    <span className="title">{`${todo.title} : `}</span>
+    <span className="status">
+      {todo.completed
+        ? <span className="completed">completed</span>
+        : <span className="not-completed">not completed</span>
+      }
```
Comment (VinnieJ-2k20) [inline src/components/Todo/Todo.js:15]: Don't render `span` inside `span`. I suggest using `classnames` library for such situations.

File: src/components/TodoList/TodoList.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Todo from '../Todo/Todo';
+import './TodoList.css';
+
+const TodoList = ({ todos }) => (
+  todos.map(todo => (
+    <ul className="list">
+      <li key={todo.id} className="list__item">
+        <Todo todo={todo} />
```
Comment (VinnieJ-2k20) [inline src/components/TodoList/TodoList.js:10]: You can use spread operator here and then easily destructure `todo` inside `Todo` component.
```suggestion
        <Todo todo={...todo} />
```

File: src/components/Todo/Todo.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    </span>
+  </div>
+);
+
+Todo.propTypes = {
+  todo: PropTypes.shape({
+    title: PropTypes.string.isRequired,
+    completed: PropTypes.bool.isRequired,
+    user: PropTypes.shape({
+      name: PropTypes.string.isRequired,
```
Comment (VinnieJ-2k20) [inline src/components/Todo/Todo.js:25]: You don't use `user.name` here, so there's no need to validate it here.

**Review by VinnieJ-2k20 — CHANGES_REQUESTED** [review]
Nice solution.
Still, there are things that need to be fixed.

File: src/App.js [hunk 4]
```diff
@@ -3,20 +3,23 @@ import './App.scss';
...
+
+const preparedTodos = todos.map((todo) => {
+  const user = users.find(person => person.id === todo.userId);
+
+  const copy = { ...todo };
+
+  copy.user = user;
+
+  return copy;
+});
```
Comment (varseniuk) [inline src/App.js:16]:
```suggestion
const preparedTodos = todos.map((todo) => {
  const user = users.find(person => person.id === todo.userId);

  return {
    ...todo,
    user,
  };
});
```

You can directly return the object with everything from `todo` inside + `user` object

File: src/App.js [hunk 5]
```diff
@@ -3,20 +3,23 @@ import './App.scss';
...
-        <span>Todos: </span>
-        {todos.length}
-      </p>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
+      <h1 className="App__title">Static list of todos</h1>
+      <TodoList todosArr={preparedTodos} />
```
Comment (varseniuk) [inline src/App.js:22]:
```suggestion
      <TodoList todos={preparedTodos} />
```
better to give the names that actually means what they contains

File: src/components/Todo/Todo.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { User } from '../User/User';
+
+export const Todo = ({ title, completed, user }) => {
+  const status = (completed)
+    ? 'done'
```
Comment (varseniuk) [inline src/components/Todo/Todo.js:7]: 👍

File: src/components/TodoLisst/TodoList.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+      <li key={todo.id} className="TodoList__item">
+        <Todo {...todo} />
+      </li>
+    ))}
+  </ol>
+);
+
+TodoList.propTypes = {
+  todosArr: PropTypes.arrayOf(
+    PropTypes.shape({}),
```
Comment (varseniuk) [inline src/components/TodoLisst/TodoList.js:18]: You could describe `todo.id` here as you are ussing it here in this component

**Review by varseniuk — APPROVED** [review]
Good.
Left some comments

**Pull request closed without merging**